### PR TITLE
Improve SSH Terminal via browser PSA service by including private key authentication method

### PR DESCRIPTION
**Pull request by aminelouati**

### CODE REVIEW

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 1]
```diff
@@ -1,39 +1,37 @@
...
-    <variable name="NATIVE_SCHEDULER" value="" description="Target native scheduler (e.g., Slurm, PBS, etc)"/>
-    <variable name="NATIVE_SCHEDULER_PARAMS" value="" description="Parameters needed to submit the job to the native scheduler"/>
-    <variable name="NODE_ACCESS_TOKEN" value="" description="If not empty, the workflow tasks will be run only on nodes that contains the specified token."/>
-    <variable name="NODESOURCENAME" value="" description="If not empty, the workflow tasks will be run only on nodes belonging to the specified node source."/>
+    <variable name="SSH_TARGET_HOST" value=""  description="Depending on the USE_REGISTERED_HOST_LIST value, the SSH target host can be either provided as string or as a value selected from the list of hosts registered by ProActive Resource Manager" group="SSH Parameters" advanced="false" hidden="false"/>
+    <variable name="USE_REGISTERED_HOST_LIST" value="false" model="PA:BOOLEAN" description="If true, change the SSH_TARGET_HOST variable model to the list of hosts registered in the resource manager" group="SSH Parameters" advanced="false" hidden="false"/>
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:7]: - This variable shloud be placed before 'SSH_TARGET_HOST' as it manages it.
- I suggest to rename it 'USE_MANAGED_HOST_LIST' meaning the list of hosts managed by the RM

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 2]
```diff
@@ -1,39 +1,37 @@
...
+    <variable name="NODE_ACCESS_TOKEN" value=""  description="If not empty, the workflow tasks will be run only on nodes that contains the specified token." group="Resource Management" advanced="true" hidden="false"/>
+    <variable name="NODE_SOURCE_NAME" value=""  description="If not empty, the workflow tasks will be run only on nodes belonging to the specified node source." group="Resource Management" advanced="true" hidden="false"/>
+    <variable name="SSH_KEY_HANDLER" value="" model="PA:SPEL(variables[&#39;USE_SSH_KEY&#39;] == &#39;false&#39; ? hideVar(&#39;SSH_KEY&#39;) : showVar(&#39;SSH_KEY&#39;) &amp;&amp; (models[&#39;SSH_KEY&#39;] = &#39;PA:CREDENTIAL&#39;) instanceof T(String))"  group="SSH Parameters" advanced="false" hidden="true"/>
+    <variable name="REGISTERED_HOST_LIST_HANDLER" value="" model="PA:SPEL(variables[&#39;USE_REGISTERED_HOST_LIST&#39;] == &#39;false&#39; ? showVar(&#39;SSH_TARGET_HOST&#39;) : showVar(&#39;SSH_TARGET_HOST&#39;) &amp;&amp; (models[&#39;SSH_TARGET_HOST&#39;] = &#39;PA:MODEL_FROM_URL(${PA_SCHEDULER_REST_PUBLIC_URL}/rm/model/hosts)&#39;)  instanceof T(String))" description="" group="SSH Parameters" advanced="false" hidden="true"/>
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:19]: - The expression 'showVar(&#39;SSH_TARGET_HOST&#39;)' is repeated in both cases, I suggest to remove it. It has no effect as 'SSH_TARGET_HOST' is not hidden in any case.
- The variable  'SSH_TARGET_HOST' has no model, so we need to add a model in both cases, I suggest to add 'PA_NOT_EMPTY_STRING' in the second case.

Comment (nebilbenmabrouk) [reply SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:19]: I suggest to reorganize the order of variables by group, it is easier to understand
1- SSH Parameters
2- Deployment Parameters 
3- Native **s**cheduler (with capital letter)
4- Resource Management

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 3]
```diff
@@ -1,39 +1,37 @@
...
     <info name="bucketName" value="service-automation"/>
     <info name="workflow.icon" value="/automation-dashboard/styles/patterns/img/wf-icons/ssh.png"/>
     <info name="pca.states" value="(VOID,RUNNING)"/>
-    <info name="Documentation" value="https://github.com/butlerx/wetty"/>
-    <info name="pca.service.id" value="SSH_via_Browser"/>
-    <info name="group" value="public-objects"/>
-    <info name="NODESOURCENAME" value="$NODE_SOURCE_NAME"/>    
+    <info name="NODESOURCENAME" value="$NODE_SOURCE_NAME"/>
     <info name="NS" value="$NATIVE_SCHEDULER"/>
+    <info name="Documentation" value="https://github.com/butlerx/wetty"/>
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:30]: The documentation link should be updated with respect to the Docker image used for wetty.

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 4]
```diff
@@ -1,39 +1,37 @@
...
     <info name="NODE_ACCESS_TOKEN" value="$NODE_ACCESS_TOKEN"/>
+    <info name="pca.service.id" value="SSH_via_Browser"/>
     <info name="NS_BATCH" value="$NATIVE_SCHEDULER_PARAMS"/>
+    <info name="group" value="public-objects"/>
   </genericInformation>
   <taskFlow>
     <task name="Start_Shell_Terminal" fork="true">
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:37]: It is better to use the non forked mode for the speed of the task.

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 5]
```diff
@@ -67,6 +65,25 @@ if ("true".equalsIgnoreCase(proxified)) {
...
+
+hostname = variables.get("SSH_TARGET_HOST")
+resolveHostnameCommand = ["host", hostname]
+executeCmd = resolveHostnameCommand.execute()
+resolveHostnameResult = executeCmd.text
+if(resolveHostnameResult.contains("not found")){
+    whatIsMyIPAddress = ["curl", "-k", "ifconfig.me"]
+  executeCmd = whatIsMyIPAddress.execute()
+  systemipaddress = executeCmd.text
+    variables.put("SSH_TARGET_HOST", systemipaddress)
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:85]: The command 'curl -k ifconfig.me' will return the ip address of the node where the task is executed, not the TARGET_SSH_HOST

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 6]
```diff
@@ -80,21 +97,20 @@ echo BEGIN "$variables_PA_TASK_NAME"
 ################################################################################
 ### THIS PART IS IMAGE SPECIFIC. IF YOU NEED TO MODIFY SOMETHING, DO IT HERE ###
 # Check if the Linux machine supports AVX instructions
-DOCKER_IMAGE=svenihoney/wetty:v1.3.0
+DOCKER_IMAGE=wettyoss/wetty
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:100]: It is better to add a specific version of the Docker image, not the latest, thi way we are sure the imae does not change

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 7]
```diff
@@ -80,21 +97,20 @@ echo BEGIN "$variables_PA_TASK_NAME"
...
-	if [ ! -z "awk -F/ '$2 == "docker"' /proc/self/cgroup" ]; then
-		host_tmp=$(hostname)
+  if [ ! -z "awk -F/ '$2 == "docker"' /proc/self/cgroup" ]; then
+    host_tmp=$(hostname)
         TARGET_HOST=${host_tmp// }
     else
-    	TARGET_HOST=$variables_PA_NODE_HOST
-	fi
+      TARGET_HOST=$variables_PA_NODE_HOST
+  fi
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:111]: In this block, you removed the case when the task is executed inside a Docker container, are you sure you want to remove it?

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml (hunk 5 above)
Comment (aminelouati) [reply SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:85]: This block should be removed

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml (hunk 7 above)
Comment (aminelouati) [reply SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:111]: Here, nothing is removed

File: SSH/resources/catalog/SSH_Terminal_Via_Browser.xml [hunk 8]
```diff
@@ -1,42 +1,40 @@
...
-    <variable name="NATIVE_SCHEDULER" value="" description="Target native scheduler (e.g., Slurm, PBS, etc)"/>
-    <variable name="NATIVE_SCHEDULER_PARAMS" value="" description="Parameters needed to submit the job to the native scheduler"/>
-    <variable name="NODE_ACCESS_TOKEN" value="" description="If not empty, the workflow tasks will be run only on nodes that contains the specified token."/>
-    <variable name="NODESOURCENAME" value="" description="If not empty, the workflow tasks will be run only on nodes belonging to the specified node source."/>
-  </variables>
+    <variable name="USE_MANAGED_HOST_LIST" value="false" model="PA:BOOLEAN" description="If true, change the SSH_TARGET_HOST variable model to the list of hosts managed in the resource manager" group="SSH Parameters" advanced="false" hidden="false"/>
+    <variable name="SSH_TARGET_HOST" value="my_target_host" model="" description="Depending on the USE_MANAGED_HOST_LIST value, the SSH target host can be either provided as string or as a value selected from the list of hosts managed by ProActive Resource Manager" group="SSH Parameters" advanced="false" hidden="false"/>
```
Comment (nebilbenmabrouk) [inline SSH/resources/catalog/SSH_Terminal_Via_Browser.xml:7]: I suggest to replace 'my_target_host' with 'host.example.net_or_IP_address'

**Review by nebilbenmabrouk — APPROVED** [review]
LGTM

**Pull request merged**